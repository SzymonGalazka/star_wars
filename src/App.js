import React from 'react';
import { Suspense } from 'react';
import { NetworkErrorBoundary } from 'rest-hooks';
import { CircularProgress as Spinner } from '@material-ui/core';
import './App.css';
import MainContainer from './components/MainContainer/MainContainer.tsx';

function App() {
  return (
    <div className="App">
      <Suspense fallback={<Spinner />}>
        <NetworkErrorBoundary>
          <MainContainer />
        </ NetworkErrorBoundary>
      </Suspense>
    </div >
  );
}

export default App;
