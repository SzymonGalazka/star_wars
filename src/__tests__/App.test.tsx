import * as React from 'react';
import TestRenderer from 'react-test-renderer';

import App from '../App';

jest.mock("../components/MainContainer/MainContainer", () => () => (
    <div>MainContainer</div>
  )
);

describe("App", () => {
  it("should render App component", () => {
    const tree = TestRenderer.create(<App />);
    expect(tree).toMatchSnapshot();
  });
});