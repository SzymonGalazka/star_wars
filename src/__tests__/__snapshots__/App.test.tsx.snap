// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render App component 1`] = `
<div
  className="App"
>
  <div
    className="MuiContainer-root MuiContainer-fixed MuiContainer-maxWidthLg"
  >
    <div
      className="MuiTypography-root MuiTypography-body1"
      style={
        Object {
          "backgroundColor": "#284f72",
          "height": "100vh",
        }
      }
    >
      <h1
        className="Header"
      >
        I want to compare two:
      </h1>
      <div
        data-testid="forces_selector"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          data-testid="Ships_button"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "margin": "10px",
              "padding": "10px",
            }
          }
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Ships
             
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-space-shuttle fa-w-20 "
              data-icon="space-shuttle"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={
                Object {
                  "margin": "5px",
                }
              }
              viewBox="0 0 640 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M592.604 208.244C559.735 192.836 515.777 184 472 184H186.327c-4.952-6.555-10.585-11.978-16.72-16H376C229.157 137.747 219.403 32 96.003 32H96v128H80V32c-26.51 0-48 28.654-48 64v64c-23.197 0-32 10.032-32 24v40c0 13.983 8.819 24 32 24v16c-23.197 0-32 10.032-32 24v40c0 13.983 8.819 24 32 24v64c0 35.346 21.49 64 48 64V352h16v128h.003c123.4 0 133.154-105.747 279.997-136H169.606c6.135-4.022 11.768-9.445 16.72-16H472c43.777 0 87.735-8.836 120.604-24.244C622.282 289.845 640 271.992 640 256s-17.718-33.845-47.396-47.756zM488 296a8 8 0 0 1-8-8v-64a8 8 0 0 1 8-8c31.909 0 31.942 80 0 80z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </span>
        </button>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          data-testid="People_button"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "margin": "10px",
              "padding": "10px",
            }
          }
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            People
             
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-user-astronaut fa-w-14 "
              data-icon="user-astronaut"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={
                Object {
                  "margin": "5px",
                }
              }
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M64 224h13.5c24.7 56.5 80.9 96 146.5 96s121.8-39.5 146.5-96H384c8.8 0 16-7.2 16-16v-96c0-8.8-7.2-16-16-16h-13.5C345.8 39.5 289.6 0 224 0S102.2 39.5 77.5 96H64c-8.8 0-16 7.2-16 16v96c0 8.8 7.2 16 16 16zm40-88c0-22.1 21.5-40 48-40h144c26.5 0 48 17.9 48 40v24c0 53-43 96-96 96h-48c-53 0-96-43-96-96v-24zm72 72l12-36 36-12-36-12-12-36-12 36-36 12 36 12 12 36zm151.6 113.4C297.7 340.7 262.2 352 224 352s-73.7-11.3-103.6-30.6C52.9 328.5 0 385 0 454.4v9.6c0 26.5 21.5 48 48 48h80v-64c0-17.7 14.3-32 32-32h128c17.7 0 32 14.3 32 32v64h80c26.5 0 48-21.5 48-48v-9.6c0-69.4-52.9-125.9-120.4-133zM272 448c-8.8 0-16 7.2-16 16s7.2 16 16 16 16-7.2 16-16-7.2-16-16-16zm-96 0c-8.8 0-16 7.2-16 16v48h32v-48c0-8.8-7.2-16-16-16z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </span>
        </button>
      </div>
      <div
        data-testid="none_winner"
        style={
          Object {
            "color": "#fff",
            "margin": "10px",
          }
        }
      />
      <div
        data-testid="match_draw"
        style={
          Object {
            "color": "#fff",
            "margin": "10px",
          }
        }
      />
    </div>
  </div>
</div>
`;
